pkg/kubelet/pod_workers.go

```diff
@@ -99,13 +101,25 @@ func newPodWorkers(runtimeCache kubecontainer.RuntimeCache, syncPodFn syncPodFnT
 		workQueue:                 workQueue,
 		resyncInterval:            resyncInterval,
 		backOffPeriod:             backOffPeriod,
+		podCache:                  podCache,
 	}
 }
 
 func (p *podWorkers) managePodLoop(podUpdates <-chan workUpdate) {
 	var minRuntimeCacheTime time.Time
+
 	for newWork := range podUpdates {
 		err := func() (err error) {
+			podID := newWork.pod.UID
+			if p.podCache != nil {
+				// TODO: Use the returned PodStatus and pass it to syncPod.
+				// This is a blocking call that would return only if the cache
+				// has an entry for the pod that is newer than minRuntimeCache
+				// Time. This ensures the worker doesn't start syncing until
+				// after the cache is at least newer than the finished time of
+				// the previous sync.
+				p.podCache.GetNewerThan(podID, minRuntimeCacheTime)
```

Comment:
This just blocks the worker until we are ready to sync. We can actually get the PodStatus and pass it to syncPod, but I want to avoid extra churns/conflicts so they will be addressed later.